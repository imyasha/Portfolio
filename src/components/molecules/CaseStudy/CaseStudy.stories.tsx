import { CaseStudyProps } from './CaseStudy';

import { CaseStudy } from './CaseStudy';
import { mockCaseStudy } from 'MockData';

export default {
	title: 'molecules/CaseStudy',
	component: CaseStudy,
};

export const CaseStudyStory = (args: CaseStudyProps) => <CaseStudy {...args} />;

CaseStudyStory.storyName = 'CaseStudy';
CaseStudyStory.args = {
	...mockCaseStudy,
	index: 0,
};
