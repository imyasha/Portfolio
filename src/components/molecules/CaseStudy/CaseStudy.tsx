import classNames from 'classnames';
import { ICaseStudy } from '@types';

import { Box } from 'Atoms/Box';
import { FloatingImages } from 'Atoms/FloatingImages';
import Image from 'next/image';
import Link from 'next/link';

export interface CaseStudyProps extends ICaseStudy {
	index: number;
}

const CaseStudy = ({
	slug,
	title,
	client,
	technologies,
	primaryImage,
	secondaryImages,
	index,
}: CaseStudyProps) => {
	const isEven = index % 2 === 0;

	return (
		<article className="mt-24 first-of-type:mt-0 transition md:hover:scale-[1.01]">
			<Link href={`/case-studies/${slug}`}>
				<a className="group">
					<Box>
						<div
							className={classNames('flex flex-col gap-4', {
								'md:flex-row': isEven,
								'md:flex-row-reverse': !isEven,
							})}
						>
							<div className="flex flex-col md:w-1/2 h-full">
								<h1 className="text-xl md:text-3xl font-bold mb-2">{title}</h1>
								<div className="mb-4 flex gap-2 items-center">
									<Image
										src={client.logo}
										alt={client.name}
										width={32}
										height={32}
										className="rounded-md"
									/>
									<h2 className="font-bold">{client.name}</h2>
								</div>
								<p className="text-sm max-w-xs">
									<strong>Tech used: </strong>
									{technologies.join(', ')}
								</p>
							</div>
							<div
								className={classNames(
									'-mb-12 -mx-10 md:mx-0 md:-mt-8 md:-mb-12',
									{
										'md:-mr-20': isEven,
										'md:-ml-20': !isEven,
									}
								)}
							>
								<FloatingImages
									topFloatingImage={secondaryImages[0]}
									bottomFloatingImage={secondaryImages[1]}
									mainImage={primaryImage}
									altText={title}
								/>
							</div>
						</div>
					</Box>
				</a>
			</Link>
		</article>
	);
};

export { CaseStudy };
