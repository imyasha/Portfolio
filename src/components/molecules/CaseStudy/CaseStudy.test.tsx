import { CaseStudy } from '../CaseStudy';
import { mockCaseStudy } from 'MockData';
import { render } from '@testing-library/react';

describe('CaseStudy', () => {
	it('renders correctly', () => {
		const { container } = render(<CaseStudy {...mockCaseStudy} index={0} />);
		expect(container).toMatchSnapshot();
	});
});
