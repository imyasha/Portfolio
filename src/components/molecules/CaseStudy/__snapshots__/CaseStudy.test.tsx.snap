// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaseStudy renders correctly 1`] = `
<div>
  <article
    class="mt-24 first-of-type:mt-0 transition md:hover:scale-[1.01]"
  >
    <a
      class="group"
      href="/case-studies/holland-and-barrett-subscriptions-service"
    >
      <div
        class="md:w-10/12 px-8 py-6 rounded-3xl glass-light dark:glass-dark mb-6 drop-shadow-lg hover:drop-shadow-2xl transition-all"
      >
        <div
          class="flex flex-col gap-4 md:flex-row"
        >
          <div
            class="flex flex-col md:w-1/2 h-full"
          >
            <h1
              class="text-xl md:text-3xl font-bold mb-2"
            >
              Subscriptions Service
            </h1>
            <div
              class="mb-4 flex gap-2 items-center"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2732%27%20height=%2732%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Holland & Barrett"
                  class="rounded-md"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
              <h2
                class="font-bold"
              >
                Holland & Barrett
              </h2>
            </div>
            <p
              class="text-sm max-w-xs"
            >
              <strong>
                Tech used: 
              </strong>
              React, Express, Next.js, Rest APIs, TypeScript, Redux, Node.js, Jest, Cypress
            </p>
          </div>
          <div
            class="-mb-12 -mx-10 md:mx-0 md:-mt-8 md:-mb-12 md:-mr-20"
          >
            <div
              class="relative inline-block w-auto"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27475%27%20height=%27275%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Subscriptions Service"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
              <div
                class="absolute w-1/2 top-0 left-12 md:left-20 z-30 animate-hover"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27330%27%20height=%27210%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Subscriptions Service"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <div
                class="absolute w-1/2 bottom-0 right-12 md:right-20 z-30 animate-hover animation-delay"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27330%27%20height=%27210%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Subscriptions Service"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </a>
  </article>
</div>
`;
